File: Examples/ObjC/Sources/BannerUI/Readme.m
#import "AppDelegate.h"
#import "ReadmeAction.h"
#import "ReadmeViewController.h"

@import BrazeKit;

#pragma mark - Internal

void fullScreenBanner(ReadmeViewController *viewController) {
  [(AppDelegate *)UIApplication.sharedApplication
          .delegate displayFullScreenBanner];
}

void wideBanner(ReadmeViewController *viewController) {
  [(AppDelegate *)UIApplication.sharedApplication
          .delegate displayWideBanner];
}

#pragma mark - Readme

NSString *const readme =
    @"This sample presents how to use the Braze provided Banner UI:\n"
    @"\n"
    @"- AppDelegate.{h,m}\n"
    @"  - Requesting Banners refresh\n"
    @"  - Banner UI presentation\n";

NSInteger const actionsCount = 2;

ReadmeAction const actions[] = {
  { @"Display a full-screen banner", @"",
    ^(ReadmeViewController *_Nonnull viewController) {
      fullScreenBanner(viewController);
    }
  },
  { @"Display a wide banner", @"",
    ^(ReadmeViewController *_Nonnull viewController) {
      wideBanner(viewController);
    }
  },
}
;

File: Examples/Swift/Sources/BannerUI/Readme.swift
import BrazeKit
import UIKit

let readme =
  """
  This sample demonstrates how to use the Braze provided Banner UI:

  - AppDelegate.swift:
    - Requesting Banners refresh
    - Banner UI presentation
  """

@MainActor
let actions: [(String, String, @MainActor (ReadmeViewController) -> Void)] = [
  (
    "Display a full-screen banner using UIKit",
    "",
    fullScreenBannerUIKit
  ),
  (
    "Display a full-screen banner using SwiftUI",
    "",
    fullScreenBannerSwiftUI
  ),
  (
    "Display a wide banner using UIKit",
    "",
    wideBannerUIKit
  ),
  (
    "Display a wide banner using SwiftUI",
    "",
    wideBannerSwiftUI
  ),
]

// MARK: - Internal

@MainActor
func fullScreenBannerUIKit(_ viewController: ReadmeViewController) {
  (UIApplication.shared.delegate as? AppDelegate)?.displayFullScreenBanner(framework: .uiKit)
}

@MainActor
func fullScreenBannerSwiftUI(_ viewController: ReadmeViewController) {
  (UIApplication.shared.delegate as? AppDelegate)?.displayFullScreenBanner(framework: .swiftUI)
}

@MainActor
func wideBannerUIKit(_ viewController: ReadmeViewController) {
  (UIApplication.shared.delegate as? AppDelegate)?.displayWideBanner(framework: .uiKit)
}

@MainActor
func wideBannerSwiftUI(_ viewController: ReadmeViewController) {
  (UIApplication.shared.delegate as? AppDelegate)?.displayWideBanner(framework: .swiftUI)
}
